### PR TITLE
[v2.6] Add K8s January patches

### PR DESCRIPTION
**Pull request by rayandas**
Related issue: https://github.com/rancher/rancher/issues/40211
Depends on PR https://github.com/rancher/kontainer-driver-metadata/pull/1061. Need to rebase this when PR 1061 is merged. 
This PR also includes template changes for ingress-nginx v1.5.1. Related issue: https://github.com/rancher/rancher/issues/40225 (depends on: https://github.com/rancher/ingress-nginx/pull/119)

### CODE REVIEW

**Review by kinarashah — COMMENTED** [review]
@rayandas we might want to split the PR into one for syncing all pending changes to `dev-v2.6-0123-patches` and another for Jan patches. I'll leave it up to you to decide tho. 

Holding off on reviewing this till we confirm changes for `nginx v1.5.1-rancher1` template, using your `dev-v2.7` PR for that.

**Comment by rayandas** [comment]
@kinarashah I raised https://github.com/rancher/kontainer-driver-metadata/pull/1068 to sync dev-v2.6-0123-patches with dev-v2.6. 
When that'll be merge I will just rebase this PR.

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: The actual published tag is [nginx-1.5.1-rancher2](https://hub.docker.com/layers/rancher/nginx-ingress-controller/nginx-1.5.1-rancher2/images/sha256-035e995c229de4f9b0571736c6937d8a0fbea27fa63cdf13baaad633cff5f92a?context=explore) so this needs to be updated here.

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: The actual published tag is [nginx-1.5.1-rancher2](https://hub.docker.com/layers/rancher/nginx-ingress-controller/nginx-1.5.1-rancher2/images/sha256-035e995c229de4f9b0571736c6937d8a0fbea27fa63cdf13baaad633cff5f92a?context=explore) so this needs to be updated here.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: These ranges need to be updated. We still want `nginxIngressv120Rancher2` to be used for 1.24.0-1.24.9 since the new one should apply only starting from 1.24.10.

File: rke/templates/nginx-v1.5.1-rancher1.go
Comment (snasovich) [inline rke/templates/nginx-v1.5.1-rancher1.go]: Rename the file and the template to Rancher2 since we published `nginx-1.5.1-rancher2` tag and not `-rancher1`.
Also rename `1` to `2` in all other relevant occurrences, e.g. in `templates.go`.

**Review by snasovich — DISMISSED** [review]

**Comment by rayandas** [comment]
Updated the PR to address all the changes.

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
